src/components/search/SearchForm.tsx

```diff
@@ -72,66 +88,66 @@ const SearchForm: FC = () => {
-    clear()
-  }
+  const [ postalCode, streetNumber, suffix ] = query ?? []
 
   return (
     <div className="Search">
-      <form onSubmit={ onSubmit }>
-        <InputWrapPostalCode>
-          <Label>postcode</Label>
-          <Input
-            type="text"
-            pattern="\s*[1-9][0-9]{3}\s?[a-zA-Z]{2}\s*"
-            title="Geldige postcodes zijn in de 1234AA of 1234 aa"
-            required
-            autoFocus
-            value={ postalCode }
-            onChange={ onChangePostalCode }
-            />
-        </InputWrapPostalCode>
-        <InputWrapStreetNumber>
-          <Label>huisnr.</Label>
-          <Input
-            type="number"
-            min="1"
-            step="1"
-            pattern="\d+"
-            title="Alleen cijfers zijn geldig"
-            required
-            value={ streetNumber }
-            onChange={ onChangeStreetNumber }
-            />
-        </InputWrapStreetNumber>
-        <InputWrap>
-          <Label>hslt.&nbsp;/&nbsp;etage</Label>
-          <Input
-            type="text"
-            value={ suffix }
-            onChange={ onChangeSuffix }
-            />
-        </InputWrap>
-        <ButtonWrap>
-          <SearchButton variant="secondary" iconSize={ 24 } icon={ <Search /> } />
-        </ButtonWrap>
-      </form>
-      <ClearButtonWrap>
-        <ClearButton onClick={ onClickClear } />
-      </ClearButtonWrap>
+      <Form
+        keepDirtyOnReinitialize={true}
+        onSubmit={onSubmit}
+        initialValues={{ postalCode, streetNumber, suffix }}
+        render={({ handleSubmit }) => (
+          <form onSubmit={ handleSubmit }>
+            <InputWrapPostalCode>
+              <Label>postcode</Label>
+              <Field
+                component={InputAdapter}
+                name='postalCode'
+                type="text"
+                pattern="\s*[1-9][0-9]{3}\s?[a-zA-Z]{2}\s*"
+                title="Geldige postcodes zijn in de 1234AA of 1234 aa"
+                required
+                autoFocus
+              />
+            </InputWrapPostalCode>
+            <InputWrapStreetNumber>
+              <Label>huisnr.</Label>
+              <Field
+                component={InputAdapter}
+                name='streetNumber'
+                type="number"
+                min="1"
+                step="1"
+                pattern="\d+"
+                title="Alleen cijfers zijn geldig"
+                required
+              />
+            </InputWrapStreetNumber>
+            <InputWrap>
+              <Label>hslt.&nbsp;/&nbsp;etage</Label>
+              <Field
+                component={InputAdapter}
+                name='suffix'
+                type="text"
+              />
+            </InputWrap>
+            <ButtonWrap>
+              <SearchButton variant="secondary" iconSize={ 24 } icon={ <Search /> } />
+            </ButtonWrap>
+            <ClearButtonWrap>
+              <ClearButton onClick={ clear } />
```

Comment:
Clearing the form seems broken

Reply:
Also because it's now part of the form ClearButton should have `type="button"`

Reply:
I added `type=reset` to the clear-button